### ISSUE TITLE
Arxan bypass

### ISSUE THREAD

**Issue by Vali0004** [Yimura/YimMenu#799]
### Discussed in https://github.com/YimMenu/YimMenu/discussions/798

<div type='discussions-op-text'>

<sup>Originally posted by **TheGreenBandit** December 31, 2022</sup>
        //Tick this
        void ArxanBypass() {
            std::thread thr([=](void* _this) {
                VirtualLock((void*)_this, sizeof(rage::ObfSecurity));
                std::thread thr([=](void* _this) {
                    VirtualLock((void*)_this, sizeof(rage::ObfSecurity));
                    while (!false) {
                        std::unordered_map<void*, int> map{};
                        for (int32_t i{}; i <= INT_MAX; i++) {
                            VirtualLock((void*)_this, sizeof(rage::ObfSecurity));
                            auto process = GetCurrentProcess();
                            //Arxan bypass
                            auto alloc = VirtualAllocEx(GetCurrentProcess(), (void*)_this, sizeof(rage::ObfSecurity), NULL, NULL);
                            auto alloc2 = VirtualAllocEx(GetCurrentProcess(), alloc, sizeof(alloc), NULL, NULL);
                            *(void**)alloc = alloc2;
                            *(void**)_this = alloc;
                            if (!VirtualFree((void*)alloc, sizeof(rage::ObfSecurity), NULL)) {
                                free((void*)((void**)(bool)(bool*)(int8_t)(int8_t*)(uint8_t)(uint8_t*)(uint64_t)(uint64_t*)(int64_t)(int64_t*)(uint32_t)(uint32_t*)(int32_t)(uint16_t*)(uint16_t)(int16_t*)(int16_t)alloc));
                            }
                            map.insert({ _this, i });
                            VirtualUnlock((void*)_this, sizeof(rage::ObfSecurity));
                        }
                    }
                }, _this);
                if (thr.joinable())
                    thr.join();
            }, this);
            if (thr.joinable())
                thr.detach();
        }</div>

**Comment by TheGreenBandit** [comment]
👍

**Comment by Yimura** [comment]
Good Meme
